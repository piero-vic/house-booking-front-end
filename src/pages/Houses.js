import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { VscTriangleLeft, VscTriangleRight } from 'react-icons/vsc';
import { useEffect, useState, useRef } from 'react';
import { displayHouses } from '../redux/actions/houses';
import Social from '../components/Social';
import PageHeading from '../components/PageHeading';

const Houses = () => {
  const houses = useSelector((state) => state.houses);
  const [slideIndex, setSlideIndex] = useState(0);
  const dispatch = useDispatch();
  const lenghtHouses = houses.length;
  const houseCard = useRef();
  const size = 250;

  useEffect(() => {
    if (houses.length === 0) {
      dispatch(displayHouses());
    }
  }, []);

  const nextSlide = () => {
    if (slideIndex === lenghtHouses - 1) return;
    const current = slideIndex + 1;
    houseCard.current.style.transform = `translateX(${-size * current}px)`;
    setSlideIndex(current);
  };

  const prevSlide = () => {
    if (slideIndex === 0) return;
    const current = slideIndex - 1;
    houseCard.current.style.transform = `translateX(${-size * current}px)`;
    setSlideIndex(current);
  };

  return (
    <div className="container relative h-screen pt-20 sm:pt-32">
      <PageHeading title="YOUR FAVOURITE HOUSES" subtitle="Please select a house model" />

      <div className="flex gap-7 overflow-hidden w-72 sm:max-w-screen-lg mx-auto sm:w-4/5">
        <div ref={houseCard} className="flex gap-7 w-full">
          {
           houses.map((house) => (
             <Link key={house.id} to={`/houses/${house.id}`}>
               <div className="flex flex-col items-center">
                 <div className="w-72">
                   <img className="rounded-md" src={house.image} alt="house-img" />
                 </div>
                 <div className="flex flex-col items-center">
                   <p className="text-base font-bold mt-3 mb-4">
                     Address:
                     {house.address}
                   </p>
                   <div className="flex gap-4 justify-center">
                     <p className="text-base font-bold  text-neutral-300">
                       Price:
                       {house.price}
                       $
                     </p>
                     <p className="text-base font-bold  text-neutral-300">
                       City:
                       {house.city}
                     </p>
                   </div>
                 </div>
                 <Social />
               </div>
             </Link>
           ))
        }
        </div>
        <button
          type="button"
          className="bg-lime-400 slider-buttons rounded-tl-3xl rounded-bl-3xl absolute right-0 top-1/3 flex items-center pl-2 sm:pl-4 w-9 h-6 sm:w-20 sm:h-11"
          onClick={nextSlide}
        >
          <VscTriangleRight className="text-white brightness-50 opacity-15" />
        </button>
        <button type="button" onClick={prevSlide} className="bg-slate-300 slider-buttons justify-end flex items-center rounded-br-3xl rounded-tr-2xl absolute top-1/3 pr-2 sm:pr-4 w-9 h-6 left-0 sm:w-20 sm:h-11">
          <VscTriangleLeft className="brightness-50 opacity-15 text-white " />
        </button>
      </div>

    </div>
  );
};

export default Houses;
