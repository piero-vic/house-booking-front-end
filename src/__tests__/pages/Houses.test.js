import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Houses from '../../pages/Houses';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Houses />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
