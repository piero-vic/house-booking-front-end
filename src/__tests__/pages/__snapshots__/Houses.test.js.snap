// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container relative h-screen pt-20 sm:pt-32"
>
  <div
    className="mb-5"
  >
    <h2
      className="text-center text-black font-black text-3xl"
    >
      YOUR FAVOURITE HOUSES
    </h2>
    <p
      className="text-center text-gray-500 font-semibold text-1xl"
    >
      Please select a house model
    </p>
  </div>
  <div
    className="flex gap-7 overflow-hidden w-72 sm:max-w-screen-lg mx-auto sm:w-4/5"
  >
    <div
      className="flex gap-7 w-full"
    />
    <button
      className="bg-lime-400 slider-buttons rounded-tl-3xl rounded-bl-3xl absolute right-0 top-1/3 flex items-center pl-2 sm:pl-4 w-9 h-6 sm:w-20 sm:h-11"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="text-white brightness-50 opacity-15"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.56 14L5 13.587V2.393L5.54 2 11 7.627v.827L5.56 14z"
        />
      </svg>
    </button>
    <button
      className="bg-slate-300 slider-buttons justify-end flex items-center rounded-br-3xl rounded-tr-2xl absolute top-1/3 pr-2 sm:pr-4 w-9 h-6 left-0 sm:w-20 sm:h-11"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="brightness-50 opacity-15 text-white "
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.44 2l.56.413v11.194l-.54.393L5 8.373v-.827L10.44 2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
